Import of kubernetes service connection results in a crash of the provider

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and Azure DevOps Provider) Version

Terraform v1.13.0
microsoft/azuredevops v1.11.2

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azuredevops_serviceendpoint_kubernetes`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

import {
  id = "${data.azuredevops_project.project.project_id}/ourResourceId"
  to = module.azdo_service_connection_kubernetes_deploy.azuredevops_serviceendpoint_kubernetes.k8s_service_connection
}

```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
Stack trace from the terraform-provider-azuredevops_v1.11.2 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 59 [running]:
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/serviceendpoint.flattenServiceEndpointKubernetes(0xc0004b0c00, 0xc000740000)
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_kubernetes.go:371 +0xcaf
github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/serviceendpoint.resourceServiceEndpointKubernetesRead(0xc0004b0c00, {0xef93a0?, 0xc0001e21e0})
	github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_kubernetes.go:212 +0x137
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x12d3de8?, {0x12d3de8?, 0xc0002c9a40?}, 0xd?, {0xef93a0?, 0xc0001e21e0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:843 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0003d9100, {0x12d3de8, 0xc0002c9a40}, 0xc0005929c0, {0xef93a0, 0xc0001e21e0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1149 +0x53a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00037a6d8, {0x12d3de8?, 0xc0002c9980?}, 0xc00016a380)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:739 +0x6c5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00027b900, {0x12d3de8?, 0xc0002c9140?}, 0xc0000fc0e0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:784 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1114520, 0xc00027b900}, {0x12d3de8, 0xc0002c9140}, 0xc00039d080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:575 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d6a00, {0x12d3de8, 0xc0002c90e0}, 0xc000116a20, 0xc0003fb9e0, 0x1a13b28, 0x0)
	google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc0000d6a00, {0x12d4718, 0xc000514000}, 0xc000116a20)
	google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 35
	google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-azuredevops_v1.11.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

<!--- What should have happened? --->

It should have imported the service connection.

### Actual Behavior

<!--- What actually happened? --->

It crashes with a panic

panic: runtime error: index out of range [0] with length 0

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. Have an existing Azure DevOps service connection for k8s.
2. Add an import block
3. `terraform apply`
### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_kubernetes.go
+++ b/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_kubernetes.go
@@ -367,36 +367,33 @@ func flattenServiceEndpointKubernetes(d *schema.ResourceData, serviceEndpoint *s
 		d.Set("azure_subscription", configItemList)
 	case "Kubeconfig":
 		var kubeconfig map[string]interface{}
-		kubeconfigSet := d.Get("kubeconfig").([]interface{})
-		configuration := kubeconfigSet[0].(map[string]interface{})
 
-		if len(configuration) > 0 {
-			kubeconfig = map[string]interface{}{}
+		kubeconfig = map[string]interface{}{}
 
-			if v, ok := configuration["kube_config"]; ok {
+		if kubeconfigSet := d.Get("kubeconfig").([]interface{}); len(kubeconfigSet) != 0 {
+			if v, ok := kubeconfigSet[0].(map[string]interface{})["kube_config"]; ok {
 				kubeconfig["kube_config"] = v.(string)
 			}
-
-			if serviceEndpoint.Data != nil {
-				if v, ok := (*serviceEndpoint.Data)["acceptUntrustedCerts"]; ok {
-					acceptUntrustedCerts, err := strconv.ParseBool(v)
-					if err != nil {
-						return fmt.Errorf("failed to parse `accept_untrusted_certs`: %+v ", err)
-					}
-					kubeconfig["accept_untrusted_certs"] = acceptUntrustedCerts
-				}
-			}
-
-			if serviceEndpoint.Authorization != nil && serviceEndpoint.Authorization.Parameters != nil {
-				if v, ok := (*serviceEndpoint.Authorization.Parameters)["clusterContext"]; ok {
-					kubeconfig["cluster_context"] = v
-				}
-			}
-
-			kubeconfigList := make([]map[string]interface{}, 1)
-			kubeconfigList[0] = kubeconfig
-			d.Set("kubeconfig", kubeconfigList)
 		}
+
+		if serviceEndpoint.Data != nil {
+			if v, ok := (*serviceEndpoint.Data)["acceptUntrustedCerts"]; ok {
+				acceptUntrustedCerts, err := strconv.ParseBool(v)
+				if err != nil {
+					return fmt.Errorf("failed to parse `accept_untrusted_certs`: %+v ", err)
+				}
+				kubeconfig["accept_untrusted_certs"] = acceptUntrustedCerts
+			}
+		}
+
+		if serviceEndpoint.Authorization != nil && serviceEndpoint.Authorization.Parameters != nil {
+			if v, ok := (*serviceEndpoint.Authorization.Parameters)["clusterContext"]; ok {
+				kubeconfig["cluster_context"] = v
+			}
+		}
+		kubeconfigList := make([]map[string]interface{}, 1)
+		kubeconfigList[0] = kubeconfig
+		d.Set("kubeconfig", kubeconfigList)
 	case "ServiceAccount":
 		var serviceAccount map[string]interface{}
 		serviceAccountSet := d.Get("service_account").([]interface{})

